### PR TITLE
Fixes #861 : JSON API for Badges Generated

### LINKED ISSUE #861: Enhancing the schemas and adding more endpoints
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

Since we are now using JSONAPI, we can add more endpoints to the current 
We can add more layers of abstractions to the schemas.

** Yes I will work on the same**

### PR DESCRIPTION
**Pull request by gabru-md**
Fixes #861 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Adds JSON API endpoint for generated badges.

Cheers!

### CODE REVIEW

**Comment by gabru-md** [comment]
<a href="https://i.imgur.com/Cakludw.png">
  
</a>

<p>
Here is an image showing the data being written into the database along side the JSON API output being produced after the badges are being produced.
</p>

Cheers!

@djmgit @ParthS007 @yashLadha @harshit98 
@vaibhavsingh97  please review

File: backend/blueprint/api/controllers/generateBadges.py [hunk 1]
```diff
@@ -21,14 +20,15 @@ def generateBadges():
                 str(e),
                 'Could not find any JSON'))
 
-    if not data.get('csv'):
+    if not data['csv']:
```
Comment (djmgit) [inline backend/blueprint/api/controllers/generateBadges.py]: will this not raise exception if csv is not present?

File: backend/blueprint/api/controllers/generateBadges.py [hunk 2]
```diff
@@ -21,14 +20,15 @@ def generateBadges():
...
         return jsonify(
             Response(401).generateMessage(
                 'No CSV filename found'))
-    if not data.get('image'):
+    if not data['image']:
```
Comment (djmgit) [inline backend/blueprint/api/controllers/generateBadges.py]: same here for image

File: backend/blueprint/api/controllers/generateBadges.py (hunk 1 above)
Comment (gabru-md) [reply backend/blueprint/api/controllers/generateBadges.py]: `.get` did not seem to work on my local. I had to make these changes to make it work.
So I changed it to this.

**Comment by gabru-md** [comment]
@djmgit can this please be merged

**Pull request merged**